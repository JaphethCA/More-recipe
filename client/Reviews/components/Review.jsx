import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import '../reviews.scss';


const propTypes = {
  review: PropTypes.objectOf(PropTypes.any).isRequired
};

/**
 * @description displays single review
 * @returns {ReactElement} markup
 */
const Review = ({ review }) => (
  <div className="single-review z-depth-2">
    <span className="user">{review.User.username}</span>
    <span className="meta-date">{moment(review.createdAt).fromNow()}</span>
    <hr />
    <p className="review-content">
      {review.content}
    </p>
  </div>
);

Review.propTypes = propTypes;

export default Review;
