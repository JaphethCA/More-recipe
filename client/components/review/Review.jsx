import React, { Component } from 'react';
import PropTypes from 'prop-types';
import UserDetails from '../user/UserDetail';
import './reviewBox.scss';


const propTypes = {
  review: PropTypes.objectOf(PropTypes.any).isRequired
};

const Review = (props) => {
  const { review } = props;
  return (
    <div className="my-review z-depth-3">
      <div>
        <UserDetails user={review.User} />
        <span className="meta-date">{new Date(review.createdAt).toDateString()}</span>
      </div>
      <hr />
      <p>
        {review.content}
      </p>
    </div>
  );
};

Review.propTypes = propTypes;

export default Review;
