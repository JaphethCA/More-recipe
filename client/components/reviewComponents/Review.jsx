import React, { Component } from 'react';
import PropTypes from 'prop-types';
import UserDetails from '../userComponent/UserDetail';
import '../../styles/sass/reviewBox.scss';

class Review extends Component {
  render() {
    const { review } = this.props;
    return (
      <div className="my-review z-depth-3">
        <div>
          <UserDetails userId={review.userId} />
          <span className="meta-date">{new Date(review.createdAt).toDateString()}</span>
        </div>
        <hr />
        <p>
          {review.content}
        </p>
      </div>
    );
  }
}

Review.propTypes = {
  reveiw: PropTypes.object.isRequired
};

export default Review;
