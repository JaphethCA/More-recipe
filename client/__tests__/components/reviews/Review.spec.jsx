import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Review from '../../../Reviews/components/Review';
import recipesMock from '../../__mock__/recipesMock';


const props = {
  review: recipesMock.Review
};

describe('<Review /> ', () => {
  it('should mount without exploding', () => {
    const wrapper = shallow(<Review {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('p').length).toBe(1);
  });

  it('should have a ".review-content" class', () => {
    const wrapper = shallow(<Review {...props} />);
    expect(wrapper.find('.review-content').length).toBe(1);
    expect(wrapper.hasClass('single-review')).toBe(true);
  });
  it('should match match snapshot', () => {
    const wrapper = shallow(<Review {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

