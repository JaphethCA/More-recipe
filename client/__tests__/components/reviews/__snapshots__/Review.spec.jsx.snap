// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review />  should match match snapshot 1`] = `
<div
  className="single-review z-depth-2"
>
  <span
    className="user"
  />
  <span
    className="meta-date"
  >
    13 days ago
  </span>
  <hr />
  <p
    className="review-content"
  >
    this is a great recipe
  </p>
</div>
`;
